cli/xlsx2json.py

```diff
@@ -181,18 +190,96 @@ def get_rows(self):
 
         return rows
 
+    def get_project_json_data(self):
+        project_data = self.get_rows()
+        if len(project_data) > 1:
+            logging.warning(f"{PROJECT} worksheet expects a single row of info but more than one found in the file. "
+                            f"Only the first row's data will be taken into consideration")
+
+        first_row = project_data[0]
+        first_row.pop('row_num')
+
+        json_key = self.xlsx_conf[WORKSHEETS_KEY_NAME][PROJECT]
+        json_value = {self.translate_header(PROJECT, k): v for k, v in first_row.items() if v is not None}
+        return {json_key: json_value}
+
+    def get_biosample_object(self, data):
+        sample_name = self.xlsx_conf[SAMPLE][OPTIONAL_HEADERS_KEY_NAME][SAMPLE_NAME_KEY]
+        scientific_name = self.xlsx_conf[SAMPLE][OPTIONAL_HEADERS_KEY_NAME][SCIENTIFIC_NAME_KEY]
+        if sample_name not in data or scientific_name not in data:
+            raise ValueError(f'If BioSample Accession is not provided, '
+                             f'The {SAMPLE} worksheet should have {SAMPLE_NAME_KEY} and {SCIENTIFIC_NAME_KEY} populated')
+
+        # BioSample expects any of organism or species field
+        data[SPECIES] = data[scientific_name]
+        # For some of the fields, BioSample expects value in arrays
+        data = {k: [v] if k in ['title', 'description', 'taxId', 'scientificName', 'commonName', SPECIES] else v
+                for k, v in data.items()}
+
+        biosample_object = {
+            "name": data[sample_name],
+            "characteristics": data
+        }
+
+        return biosample_object
+
+    def get_sample_data_with_split_analysis_alias(self, data, analysis_alias, sample_name_in_vcf):
+        if analysis_alias not in data or sample_name_in_vcf not in data:
+            raise ValueError(
+                f'Worksheet {SAMPLE} does not have required field {ANALYSIS_ALIAS_KEY} or {SAMPLE_NAME_IN_VCF_KEY}')
```

Comment:
I think I'd prefer if that test was done next to the use of the `required` key line 249 bellow.

Reply:
done